flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java

```diff
@@ -830,53 +797,30 @@ else if (result instanceof DisposeSavepointFailure) {
 	//  Interaction with programs and JobManager
 	// --------------------------------------------------------------------------------------------
 
-	protected int executeProgramDetached(PackagedProgram program, Client client, int parallelism) {
-		LOG.info("Starting execution of program");
+	protected int executeProgram(PackagedProgram program, Client client, int parallelism) {
+		logAndSysout("Starting execution of program");
 
 		JobSubmissionResult result;
 		try {
-			result = client.runDetached(program, parallelism);
```

Comment:
It seems that the runDetached is only used in the `DetachedEnvironment`. Maybe we can use the setter-variant there as well to get rid of the method completely.